pkg/performance/types.go

```diff
@@ -368,6 +370,54 @@ func (c *CollectionConfig) ApplyDefaults() {
 	}
 }
 
+// ValidateOption specifies validation requirements for CollectionConfig
+type ValidateOption struct {
+	RequireHostProcPath bool
+	RequireHostSysPath  bool
+	RequireHostDevPath  bool
+}
+
+// Validate ensures that all configured paths are absolute paths and that required paths are non-empty.
+// This centralizes path validation logic previously duplicated across all collectors.
+func (c CollectionConfig) Validate(opts ...ValidateOption) error {
```

Comment:
why is the signature `...ValidateOption`? Wouldn't it be simpler to just have `(opt ValidationOption)`?